pymongo/read_preferences.py

```diff
@@ -103,6 +104,11 @@ def _validate_hedge(hedge: Optional[_Hedge]) -> Optional[_Hedge]:
     if not isinstance(hedge, dict):
         raise TypeError(f"hedge must be a dictionary, not {hedge!r}")
 
+    warnings.warn(
+        "Hedged reads are deprecated starting in server version 8.0.",
```

Comment:
Mentioning the server side deprecation is good but we should first mention the driver side deprecation. Something like:
`the read preference "hedge" option is deprecated in PyMongo 4.12+ because hedged reads are deprecated in MongoDB version 8.0+. Support for "hedge" will be removed in PyMongo 5.0.`